saleor/core/weight.py

```diff
@@ -98,7 +98,8 @@ def validate(self, weight):
                 raise Exception("%r is not a valid weight." % (weight,))
             if weight.unit != unit:
                 raise forms.ValidationError(
-                    "Invalid unit: %r (expected %r)." % (weight.unit, unit)
+                    "Invalid unit: %r (expected %r)." % (weight.unit, unit),
+                    code="invalid",
```

Comment:
Shouldn't we use `ErrorCode` class here?

Reply:
I cannot see any way to avoid it for now; I've created a ticket for that, as we discussed: #4745